src/server/aes_cbc_block/cbe-cxx-cxx_pool.adb

```diff
@@ -0,0 +1,130 @@
+--
+-- Copyright (C) 2019 Genode Labs GmbH, Componolit GmbH, secunet AG
+--
+-- This file is part of the Consistent Block Encrypter project, which is
+-- distributed under the terms of the GNU Affero General Public License
+-- version 3.
+--
+
+pragma Ada_2012;
+
+with CBE.Request;
+with CBE.Primitive;
+with GNAT.IO; use GNAT.IO;
+
+package body CBE.CXX.CXX_Pool
+with Spark_Mode
+is
+	--
+	-- Initialize_Object
+	--
+	procedure Initialize_Object(
+		Obj       : in out Pool.Object_Type;
+		Obj_Size  : in     CXX_Size_Type;
+		Req_Size  : in     CXX_Size_Type;
+		Prim_Size : in     CXX_Size_Type)
+	is
+		Req  : constant CXX_Request.Object_Type   := (0,0,0,0,0,0);
+		Prim : constant CXX_Primitive.Object_Type := (0,0,0,0,0);
+	begin
+		if not Fitting_Object_Size(Obj'Size, Obj_Size) then
+			raise Program_Error;
+		elsif not Correct_Object_Size(Req'Size, Req_Size) then
+			raise Program_Error;
+		elsif not Correct_Object_Size(Prim'Size, Prim_Size) then
+			raise Program_Error;
+		end if;
```

Comment:
```Ada
if
   not Fitting_Object_Size(Obj'Size, Obj_Size)
   or else not Correct_Object_Size(Req'Size, Req_Size)
   or else not Correct_Object_Size(Prim'Size, Prim_Size)
then
   raise Program_Error;
end if;
```
Is equivalent to this expression. Again this should be handled in a precondition to prove that the exceptions will never be raised.